protobuf/model_config.proto

```diff
@@ -1382,6 +1382,24 @@ message ModelSequenceBatching
     //@@     The optional field to specify the initial state for the model.
     //@@
     repeated InitialState initial_state = 5;
+
+    //@@  .. cpp:var:: bool use_single_buffer
+    //@@
+    //@@     The optional field to use a single buffer for both input and output state.
+    //@@     The default value is false.
+    //@@
+    bool use_single_buffer = 6;
+
+    //@@  .. cpp:var:: bool use_growable_memory
+    //@@
+    //@@     The optional field to allow an implicit state buffer to grow or shrink
+    //@@     when the size changes during a sequence. When using this option Triton
+    //@@     guarantess that it will use the same buffer even if the state size changes.
+    //@@     Currently, this option only applies for implicit state that uses CUDA and
+    //@@     use_single_buffer must be enabled.
```

Comment:
IIUC, we force the implicit state memory used in growable block manager to be GPU memory for now. Maybe the wording here could be clearer that regardless of what the user does, this will use GPU memory for now, even if they request other memory types.

Reply:
Why wouldn't we error if they try different types? Just double checking - we don't silently switch preferences, correct?

Reply:
>Why wouldn't we error if they try different types?

What the user specifies is just "preferences" (i.e., there is no guarantee that Triton will satisfy the request). The same pattern exists in `TRITONBackend_OutputBuffer`.

Reply:
I guess the question is which options take precedence - does the model config or the API call to get the buffer - will discuss offline